### PR TITLE
resolve several null pointer dereferences

### PR DESCRIPTION
**Pull request by chipitsine**
those issues were found by cppcheck

### CODE REVIEW

File: MagickCore/nt-base.c [hunk 1]
```diff
@@ -1793,20 +1793,20 @@ MagickPrivate DIR *NTOpenDirectory(const char *path)
     {
```
Comment (dlemstra) [inline MagickCore/nt-base.c]: We should remove the null check here instead. `AcquireCriticalMemory` will never be null.

File: coders/msl.c [hunk 2]
```diff
@@ -552,7 +552,8 @@ static void MSLPushImage(MSLInfo *msl_info,Image *image)
   ssize_t
     n;
 
-  (void) LogMagickEvent(TraceEvent,GetMagickModule(),"%s",image->filename);
+  if (image != (Image *) NULL)
```
Comment (dlemstra) [inline coders/msl.c]: Good catch!

File: MagickCore/montage.c [hunk 3]
```diff
@@ -412,9 +412,9 @@ MagickExport Image *MontageImageList(const ImageInfo *image_info,
   number_images=GetImageListLength(images);
   master_list=ImageListToArray(images,exception);
   image_list=master_list;
```
Comment (dlemstra) [inline MagickCore/montage.c]: Can you move this line down also?

File: MagickCore/nt-base.c (hunk 1 above)
Comment (chipitsine) [reply MagickCore/nt-base.c]: if I will do it in new commit, can you squash all commits during merge ?

**Comment by chipitsine** [comment]
I pushed with fixes

File: MagickCore/nt-base.c (hunk 1 above)
Comment (dlemstra) [reply MagickCore/nt-base.c]: I can squash your changes.

**Comment by dlemstra** [comment]
Could you also open a PR for our [ImageMagick6](https://github.com/ImageMagick/ImageMagick6) repository to resolve these issues?

**Pull request merged**